source/boot_security.c

```diff
@@ -1,11 +1,21 @@
 #include "boot_security.h"
+#include "aes.h"
+#include "hal_log.h"
+#define EXAMPLE_AES_KEY_LEN_IN_BITS		256
+#define EXAMPLE_AESKEY_LEN_IN_BYTES		(EXAMPLE_AES_KEY_LEN_IN_BITS / 8)
+#define AES_IV_LENGTH					(16)
 
 bool boot_authenticate_upgrade_package(boot_upgrade_package_t* package)
 {
     return true;
 }
-
 void boot_decrypt_upgrade_package(boot_upgrade_package_t* package)
 {
-
-}
+    mbedtls_aes_context aes;
+    unsigned char key[EXAMPLE_AESKEY_LEN_IN_BYTES];
```

Comment:
Please do not use primitive types like "char" instead use uint8_t here which explicitly specifies the variable length (herein, 8 bits)
`
uint8_t key[EXAMPLE_AESKEY_LEN_IN_BYTES];
`

Reply:
The Key must be a global variable so do not define in the function.